src/components/BookForm.js

```diff
@@ -1,40 +1,48 @@
 import React, { useState } from 'react';
-import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
+import Button from './Button';
+
+const BookForm = () => {
+  const categories = [
+    { id: 1, name: 'Learning' },
+    { id: 2, name: 'Philosophy' },
+    { id: 3, name: 'Religion' },
+  ];
+
+  const dispatch = useDispatch();
 
-const BookForm = ({ onAdd }) => {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('Learning');
 
-  const handleSubmit = (e) => {
+  const submitAddBook = (e) => {
     e.preventDefault();
-    if (title && author) {
-      onAdd({ title, author });
-      setTitle('');
-      setAuthor('');
-    }
+    dispatch(addBook({
+      id: '4',
```

Comment:
- [ Required ] Good job adding the `add book` and `remove book` buttons as requested by the previous reviewer, however, it appears the changes introduced a new problem in your project. If I add `3` books to the book list and try to remove one of them from the list it deletes all the books from the list. This is because all the newly added books on the list have the id of `4`, so all the books that have an `id` of `4` are deleted from the `redux store`. I would like to suggest that you install the `uuid` library in your project and use it to generate id's for newly created books. Please see the example code below for an idea on how to achieve this.

## Example code
> Install uuid library in your project
```sh
  $ npm install uuid
```
> Refactor the BookForm.js file as shown below
```JavaScript
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { v4 as uuidv4 } from 'uuid'; // Import uuid library in to your project
import { addBook } from '../redux/books/booksSlice';
import Button from './Button';

const BookForm = () => {
  const categories = [
    { id: 1, name: 'Learning' },
    { id: 2, name: 'Philosophy' },
    { id: 3, name: 'Religion' },
  ];

  const dispatch = useDispatch();

  const [title, setTitle] = useState('');
  const [author, setAuthor] = useState('');
  const [category, setCategory] = useState('Learning');

  const submitAddBook = (e) => {
    e.preventDefault();
    dispatch(addBook({
      id: uuidv4(), // generate unique id when a new book is created
      title,
      author,
      progress: 'currently reading',
      categories: ['All', category],
    }));
    setTitle('');
    setAuthor('');
  };

  return (
    <form onSubmit={(e) => addBook(e)}>
      <input type="text" value={title} onChange={(e) => setTitle(e.target.value)} placeholder="Title" />
      <input type="text" value={author} onChange={(e) => setAuthor(e.target.value)} placeholder="Author" />
      <select name="categories" id="categories" onChange={(e) => setCategory(e.target.value)}>
        {categories.map((category) => (
          <option key={category.id} value={category.name}>
            {category.name}
          </option>
        ))}
      </select>
      <Button onClick={(e) => submitAddBook(e)} text="Add" />
    </form>
  );
};

export default BookForm;
```